### ISSUE TITLE
How to get the loss on the test set？

### ISSUE THREAD

**Issue by Shiguang-Guo** [facebookresearch/fairseq#4319]
I want to get the loss on the test set. I didn't find the code for generating the loss part in the generate.py file, and the part about generating the loss in the training file is too complicated. Is there an elegant way to get the loss on the test set?

#### What's your environment?

 - fairseq Version : 0.9.0
 - PyTorch Version：1.8
 - OS (e.g., Linux):Linux

**Comment by gmryu** [comment]
Can you try fairseq-eval-lm: https://fairseq.readthedocs.io/en/latest/command_line_tools.html#fairseq-eval-lm ?
It basically uses `--path CHECKPOINT --gen-subset WHICH_DATA`  and other common arguments like fp16.
I remember using this for my transformer a while ago.(Faint memory and I regret I haven't left notes).
So if you happen to succeed, please let me know as well.

Just a note, generation's cross entropy loss does not exist because generated outputs' length is different than answers' and cross entropy needs two sequence to have the same length.
In other words, generate.py cannnot compute loss in its nature.
